### PR TITLE
Docs: version bumps for 1.15 release

### PR DESCRIPTION
**Pull request by Rachael-Graham**
Bumps active versions and version conrefs in the docs for 1.15.0 release.

### CODE REVIEW

File: docs/active_versions.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
-  "latest": "v1.14.x",
+  "latest": "v1.15.x",
```
Comment (sam-heilbron) [inline docs/active_versions.json:2]: Let's keep 1.14 as the default "latest" for now, and we will likely switch this in a week/two

**Review by sam-heilbron — DISMISSED** [review]

**Review by sam-heilbron — APPROVED** [review]

File: docs/active_versions.json [hunk 2]
```diff
@@ -2,11 +2,11 @@
   "latest": "v1.14.x",
```
Comment (artberger) [inline docs/active_versions.json:2]: Should this be 1.15.x now?

File: docs/active_versions.json [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
   "versions": [
     "main",
+    "v1.15.x",
     "v1.14.x"
```
Comment (artberger) [inline docs/active_versions.json:6]: Should 1.14.x be moved to "oldVersions"?

Comment (Rachael-Graham) [reply docs/active_versions.json:6]: Threw an error when I did, so I have to keep it in here as long as it is latest still

File: docs/active_versions.json (hunk 2 above)
Comment (Rachael-Graham) [reply docs/active_versions.json:2]: Request from Sam to keep it as 1.14 for now

**Review by artberger — APPROVED** [review]

**Pull request merged**